sympy/physics/units/tests/test_dimensionless.py

```diff
@@ -0,0 +1,105 @@
+from sympy.physics.units import Dimension
+from sympy.physics.units import DimensionSystem
+from sympy import Rational
+"""
+We need to define the base quantities globally to work with and
+define the derived quantities from there on
+"""
+
+M = Dimension('mass')
+L = Dimension('length')
+T = Dimension('time')
+I = Dimension('current')
+
+dummy_si_system = DimensionSystem((L, M, T, I))
+
+def test_navier_stokes_case_si():
+    """ In this case my function will be able to generate one set of
+    dimensionless quantities in Navier-Stokes equations ......"""
+    Lc = L # Characteristic length scale
+    Uc = L/T # Characteristic velocity scale
+    rho_c = M/L**3 # Characteristic density scale
+    mu = M/L/T # Dynamic viscosity
+    p_c = M/L/T**2 # Characteristic pressure scale
+    F_bc = M*L/T**2 # Characteristic body forces scale
+    list_of_quantities = {'Lc': Lc, 'Uc': Uc, 'rho': rho_c, 'mu': mu, 'pc': p_c, 'Fbc': F_bc}
```

Comment:
why create a dictionary? What about just passing a list instead?

Reply:
Hi Francesco, @Upabjojr Actually I tried a lot on this, but I found several hurdles along this way: The resulting Dimension class either comes out to be 1 (since the input quantities are derived quantities and these are sort of stored in the base forms only) or an equivalent radical which eventually is equivalent to one. Thus I could not find in any equivalent list based representation that worked. The second one being that there are some derived quantities such as momentum (for example, there is Planck's momentum in SI dimension system in the package but not many may ask for it directly), viscosity, elasticity, gauge field units etc. which may be of explicit importance to the users. Thus I think it's appropriate to use the dictionary where the user is allowed to define the name of the quantity of their own choice, and tuple it with the corresponding dimensions.